### PR TITLE
DIT-409 adding integration tests to cover mandatory/extra/all fields …

### PR DESCRIPTION
**Pull request by jumpertester**
…when creating a case.

### CODE REVIEW

File: app/uk/gov/hmrc/bindingtariffclassification/todelete/CaseData.scala [hunk 1]
```diff
@@ -24,12 +24,28 @@ import uk.gov.hmrc.bindingtariffclassification.utils.RandomGenerator
 object CaseData {
 
   def createBTIApplication: BTIApplication = {
```
Comment (ed0906) [inline app/uk/gov/hmrc/bindingtariffclassification/todelete/CaseData.scala]: Could we name this something like `createBasicBTIApplication` or something along those lines so it's clear we're working with the bare-bones one

File: test/it/uk/gov/hmrc/component/CaseSpec.scala [hunk 2]
```diff
@@ -89,6 +95,42 @@ class CaseSpec extends BaseFeatureSpec {
...
+        .postData(c3Json.toString()).asString
+
+      Then("The response code should be created")
+      result.code shouldEqual CREATED
+
+      And("The case is returned in the JSON response")
+      val responseCase = Json.parse(result.body).as[Case]
+      responseCase.reference shouldBe "1"
+      responseCase.status shouldBe CaseStatus.NEW
+      responseCase.createdDate should not be ZonedDateTime.parse("2000-01-02T00:00:00+01:00[Europe/Paris]")
```
Comment (ed0906) [inline test/it/uk/gov/hmrc/component/CaseSpec.scala]: I think we have a matcher in the NewCastRequestSpec that lets you specify a rough date you could use for this

Comment (ed0906) [reply test/it/uk/gov/hmrc/component/CaseSpec.scala]: It ends up something like `responseCase.createdDate shouldBe roughly(ZonedDateTime.now())`

File: app/uk/gov/hmrc/bindingtariffclassification/todelete/CaseData.scala (hunk 1 above)
Comment (jumpertester) [reply app/uk/gov/hmrc/bindingtariffclassification/todelete/CaseData.scala]: Updated name to '''createBasicBTIApplication'''

File: test/it/uk/gov/hmrc/component/CaseSpec.scala (hunk 2 above)
Comment (jumpertester) [reply test/it/uk/gov/hmrc/component/CaseSpec.scala]: Refactored the matcher to live in a util object within the test folder and used it for the above.

**Pull request merged**